app/parse.py

```diff
@@ -8,5 +15,39 @@ class Course:
     duration: str
 
 
+FIELDS_LIST = [field.name for field in fields(Course)]
+
+
+def parse_single_course(soup: Tag) -> Course:
+    return Course(
+        name=soup.select_one("a.ProfessionCard_title__fEWio > h3").text,
+        short_description=soup.select_one(
+            ".ProfessionCard_professionTags__2iarD + "
+            "p.typography_landingTextMain__Rc8BD"
+        ).text,
+        duration=soup.select_one(
+            "p.ProfessionCardTags_regularTag__dqOGj:last-child span"
+        ).text
+    )
+
+
 def get_all_courses() -> list[Course]:
-    pass
+    text = requests.get(BASE_URL).content
+    soup = BeautifulSoup(text, "html.parser")
+    courses = soup.select(".ProfessionCard_cardWrapper__2Q8_V")
+    return [parse_single_course(course) for course in courses]
+
+
+def write_to_csv(courses: list[Course]) -> None:
+    with open("courses.csv", "w") as f:
```

Comment:
It's a good practice to specify the newline parameter in the `open` function when writing CSV files to prevent extra blank lines on some platforms. Use `open('courses.csv', 'w', newline='')`.